### PR TITLE
chore(storage): remove sinon as a dependency; convert sinon spies to mock functions

### PR DESCRIPTION
**Pull request by calebpollman**
#### Description of changes

Remove `sinon` as a dependency of `@aws-amplify/storage` by converting `sinon` spies to `jest` mock functions.

> There are some additional updates due to `prettier`, relevant changes are noted.

#### Issue #, if available

https://github.com/aws-amplify/amplify-js/issues/9565

#### Description of how you validated changes

`yarn clean && yarn && yarn bootstrap && yarn build && yarn test`

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/storage/__tests__/providers/AWSS3ProviderManagedUpload-unit-test.ts [hunk 1]
```diff
@@ -79,61 +78,88 @@ afterEach(() => {
...
+		);
 	});
 });
 
 describe('multi part upload tests', () => {
 	test('happy case: upload a string as body that splits in two parts', async () => {
 		// setup event handling
 		const emitter = new events.EventEmitter();
-		const eventSpy = sinon.spy();
+		const eventSpy = jest.fn();
```
Comment (calebpollman) [inline packages/storage/__tests__/providers/AWSS3ProviderManagedUpload-unit-test.ts:146]: Replace `sinon` spy with `jest` mock function

File: packages/storage/__tests__/providers/AWSS3ProviderManagedUpload-unit-test.ts [hunk 2]
```diff
@@ -187,14 +213,14 @@ describe('multi part upload tests', () => {
 
 		// Progress report testing
 		// First progress is reported at the end, when first full part is uploaded
-		expect(eventSpy.getCall(0).args[0]).toStrictEqual({
+		expect(eventSpy).toHaveBeenNthCalledWith(1, {
```
Comment (calebpollman) [inline packages/storage/__tests__/providers/AWSS3ProviderManagedUpload-unit-test.ts:216]: update usage of `expect(...).toStrictEqual` with `expect(...).toHaveBeenNthCalledWith`

File: packages/storage/__tests__/providers/AWSS3ProviderManagedUpload-unit-test.ts [hunk 3]
```diff
@@ -205,35 +231,37 @@ describe('multi part upload tests', () => {
 	test('error case: upload a string as body that splits in two parts but second part fails', async () => {
 		// setup event handling
 		const emitter = new events.EventEmitter();
-		const eventSpy = sinon.spy();
+		const eventSpy = jest.fn();
```
Comment (calebpollman) [inline packages/storage/__tests__/providers/AWSS3ProviderManagedUpload-unit-test.ts:234]: Replace `sinon` spy with `jest` mock function

File: packages/storage/__tests__/providers/AWSS3ProviderManagedUpload-unit-test.ts [hunk 4]
```diff
@@ -287,13 +315,13 @@ describe('multi part upload tests', () => {
 			UploadId: testUploadId,
 		});
 		// Progress reporting works as well
-		expect(eventSpy.getCall(0).args[0]).toStrictEqual({
+		expect(eventSpy).toHaveBeenNthCalledWith(1, {
```
Comment (calebpollman) [inline packages/storage/__tests__/providers/AWSS3ProviderManagedUpload-unit-test.ts:318]: update usage of `expect(...).toStrictEqual` with `expect(...).toHaveBeenNthCalledWith`

**Comment by jamesaucode** [comment]
LGTM!

**Review by jamesaucode — APPROVED** [review]

**Review by katiegoines — APPROVED** [review]

**Pull request merged**